### ISSUE TITLE
[Feature] Add lambda function expressions

### ISSUE THREAD

**Issue by lorenzholzbauer** [Kipper-Lang/Kipper#572]
### Is there an existing proposal for this?

- [X] I have searched the existing issues

### This feature does not exist in the latest version

- [X] I am using the latest version

### Proposal

Add lambda function support, which allows users to specify anonymous functions without a name that always return a value.

For example:
```js
(x: num) => x + 1;

// Once generics are implemented with #
var y: Func<num, num> = (x: num) => x + 1;
```

### Exact behaviour / changes you want

- [x] Add lambda notation and semantic analysis.
- [x] Add translation to JavaScript and TypeScript.
- [x] Add tests

**Comment by Luna-Klatzer** [comment]
Generic type support will be implemented by #494 and #584.

**Comment by Luna-Klatzer** [comment]
Implemented by #650.
